cpp/src/arrow/util/bitmap_test.cc

```diff
@@ -0,0 +1,64 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <gtest/gtest.h>
+
+#include "arrow/testing/gtest_util.h"
+#include "arrow/util/bitmap_ops.h"
+
+namespace arrow::internal {
+
+TEST(BitmapOpsTest, PartialLeadingByteSafety) {
```

Comment:
Since `CopyBitmap` and others are tested in `bit_util_test.cc`, should we move those tests here?

Reply:
Yeah, saw the bunch of cases in `bit_util_test.cc` are actually about `bitmap_ops.cc`. Plus the `bit_util_test.cc` are already over 2500 LOC. Will do.